### ISSUE TITLE
Fastar for custom dataset

### ISSUE THREAD

**Issue by justin020025** [vsahil/FastAR-RL-for-generating-AR#2]
Hi, I really appreciated your work that made AR efficient. Could you give a brief guide to implementing FASTAR on our own dataset? There seems to be some pieces of codes that I should modified, not sure where to start. Also, I wonder if the categorical features are integer-encoded in this implementation? Thanks a lot.

**Comment by vsahil** [comment]
Hello, you need to start by creating a gym environment for your dataset here: [fastar]/[gym-midline]/[gym_midline]/envs/ (like the way it is defined for the other datasets), and then you need to train a model and get ARs (instructions for which are given in the Readme file)

**Comment by justin020025** [comment]
Thank you so much. I'm trying to deal with categorical features. So I should turn each category to an integer, and the action space still remains as gym.spaces.Discrete(2 * len(dataset.columns)). Am I correct on this?

**Comment by vsahil** [comment]
Yes you are right. This is assuming that each categorical feature is ordinal. You can also encode them one-hot, and then ensure that only one of the one-hot encoded rows have a 1.

**Comment by justin020025** [comment]
I prefer one-hot since my classifier is trained on one-hot encoded. So now I should increase the size of gym.spaces.Discrete() right? And to ensure only one of the one-hot encoded rows have a 1, I'm think about taking Argmax over the rows after finishing. Do you have any other suggestion? Thanks.

**Comment by vsahil** [comment]
Yes, increase the gym.space.Discrete size -- equal to the number of total feature after one-hot encoding. yeah argmax should be good -- ensure all other are set to 0 and the argmax column to 1.

**Comment by justin020025** [comment]
Hi, I noticed that in F.1 of appendix in the paper, you mentioned that the action space is discretized in the comparison with random and greedy search. Does this mean there's another implementation that support continuous action space?
